posthog/hogql/metadata.py

```diff
@@ -28,10 +31,22 @@ def get_hogql_metadata(
             translate_hogql(query.expr, context=context)
         elif isinstance(query.select, str):
             context = HogQLContext(team_id=team.pk, enable_select_queries=True)
+            context.database = create_hogql_database(context.team_id)
             select_ast = parse_select(query.select)
             _is_valid_view = is_valid_view(select_ast)
-            response.isValidView = _is_valid_view
-            print_ast(parse_select(query.select), context=context, dialect="clickhouse")
+
+            # Kludge: redundant pass through the AST (called in print_ast)
```

Comment:
I think this is ok as it's a simple pass and I'd rather do this than fiddle with existing traversers that might be reused elsewhere

Reply:
What does "called in print_ast" mean? The pass is happening two lines below, no? 🤔

Reply:
Sorry, confusing comment. Just meant that we're doing another extra traversal because print_ast does a traversal

Reply:
Can you perhaps clarify the comment? :P